velox/exec/Spill.cpp

```diff
@@ -430,6 +435,84 @@ uint32_t SpillPartitionId::encodedId() const {
   return encodedId_;
 }
 
+SpillPartitionIdLookup::SpillPartitionIdLookup(
+    const SpillPartitionIdSet& spillPartitionIds,
+    uint32_t startPartitionBit,
+    uint32_t numPartitionBits)
+    : partitionBitsMask_(
+          bits::lowMask(
+              numPartitionBits * (SpillPartitionId::kMaxSpillLevel + 1))
+          << startPartitionBit) {
+  VELOX_CHECK(!spillPartitionIds.empty());
+
+  const auto numLookupBits =
+      (SpillPartitionId::kMaxSpillLevel + 1) * numPartitionBits;
+  VELOX_CHECK_LT(
+      startPartitionBit,
+      sizeof(uint64_t) * 8 - numLookupBits,
+      "Insufficient lookup bits.");
+  lookup_.resize(1UL << numLookupBits, SpillPartitionId::kEmptyId);
+
+  for (const auto& id : spillPartitionIds) {
+    generateLookup(id, startPartitionBit, numPartitionBits);
+  }
+}
+
+void SpillPartitionIdLookup::generateLookup(
```

Comment:
This can be static method? Not sure if we want to compute the max spill level in spillPartitionIds to save space? Otherwise we could std::array as its size is prior known.